### PR TITLE
[Model Monitoring] Replace `endpoint_id` with `uid` to handle old model endpoint schemas

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
This PR is a fix for a backwards compatibility issue when calling `get_model_endpoint` API - https://jira.iguazeng.com/browse/ML-3866.

Following #2685, the key name of the unique model endpoint id in the database is stored as `uid` instead of `endpoint_id`. However, if a model endpoint record has been stored in the KV table before #2685 (`<1.3.1`) it still has the old key name - `endpoint_id`. As a result, the `uid` key has been missing from the result of `get_model_endpoint` API  when trying to get an old model endpoint record. In this PR, we fix that issue by replacing the key name `endpoint_id` with `uid`.

### CODE REVIEW

File: mlrun/utils/model_monitoring.py [hunk 1]
```diff
@@ -241,3 +241,10 @@ def validate_errors_and_metrics(endpoint: dict):
                 }
             }
         )
+    # Validate key `uid` instead of `endpoint_id`
+    # For backwards compatibility reasons, we replace the `endpoint_id` with `uid` which is the updated key name
+    if model_monitoring_constants.EventFieldType.ENDPOINT_ID in endpoint:
+        endpoint[model_monitoring_constants.EventFieldType.UID] = endpoint[
+            model_monitoring_constants.EventFieldType.ENDPOINT_ID
+        ]
+        endpoint.pop(model_monitoring_constants.EventFieldType.ENDPOINT_ID)
```
Comment (theSaarco) [inline mlrun/utils/model_monitoring.py:250]: Why not keep both? Conceptually, the caller may be an old client that still expected the `ENDPOINT_ID` field in the result. We support old clients as long as the server is new (not vice-versa).

**Review by theSaarco — CHANGES_REQUESTED** [review]
I think we should keep both fields in the response, unless I'm missing something.

File: mlrun/utils/model_monitoring.py (hunk 1 above)
Comment (Eyal-Danieli) [reply mlrun/utils/model_monitoring.py]: right, we'll keep both

**Review by theSaarco — APPROVED** [review]
👍

**Pull request merged**